src/main/java/com/mael/ttt/Game.java

```diff
@@ -51,7 +55,7 @@ private void setPlayers() {
     private boolean nextTurn() {
         gameUI.printBoard(board);
 
-        Player currentPlayer = (currentMark.equals("X")) ? player1 : player2;
+        Player currentPlayer = (currentMark.equals(playerMark)) ? player1 : player2;
         int index            = currentPlayer.getCellIndex(board);
         board.setCell(index, currentMark);
 
```

Comment:
Do you mean I am holding state in the Board class?
`board.setCell(index, value)` sets the cell at `index` with `value`, which, as you said, is the players choice. I need the index to know where to save the value.